File: backends/go-gin/internal/infrastructure/cache.go
package infrastructure

import (
	"context"
	"log"

	"github.com/go-redis/redis/v8"
)

func NewRedis(url string) *redis.Client {
	opt, err := redis.ParseURL(url)
	if err != nil {
		log.Fatalf("Invalid Redis URL: %v", err)
	}

	rdb := redis.NewClient(opt)
	if err := rdb.Ping(context.Background()).Err(); err != nil {
		log.Fatalf("Failed to connect to Redis: %v", err)
	}

	return rdb
}

File: backends/rust-axum/src/infrastructure/cache.rs
use redis::aio::MultiplexedConnection;

pub async fn create_client(redis_url: &str) -> MultiplexedConnection {
    let client = redis::Client::open(redis_url).expect("Invalid Redis URL");
    client
        .get_multiplexed_async_connection()
        .await
        .expect("Failed to connect to Redis")
}